Makefile

```diff
@@ -11,10 +16,18 @@ GCOVR := gcovr
 override LDLIBS += -levent -lhiredis -lswsscommon -pthread -lboost_thread -lboost_system
 override CPPFLAGS += -Wall -std=c++17 -fPIE -I/usr/include/swss
 override CPPFLAGS += -MMD -MP -MF"$(@:%.o=%.d)" -MT"$(@)"
-CPPFLAGS_TEST := --coverage -fprofile-arcs -ftest-coverage -fprofile-generate -fsanitize=address
-LDLIBS_TEST := --coverage -lgtest -pthread -lstdc++fs -fsanitize=address
+CPPFLAGS_TEST := -I$(GMOCK_GLOBAL_INC_PATH) --coverage -fprofile-arcs -ftest-coverage -fprofile-generate -fsanitize=address
+LDLIBS_TEST := --coverage -lgtest -lgmock -pthread -lstdc++fs -fsanitize=address
 PWD := $(shell pwd)
 
+$(addprefix $(DEST)/, $(BUILD_TEST_DIR)): $(DEST)/% :
+	if [ ! -d ${GMOCK_GLOBAL_DIR} ]
+	then
+		git clone https://github.com/apriorit/gmock-global.git $(GMOCK_GLOBAL_DIR)
+	fi
+	sudo cp -r $(GMOCK_GLOBAL_DIR)/include/gmock-global /usr/include/
```

Comment:
Instead of doing this, it would be far better to add that repo as a submodule to this repository. In addition, instead of copying it to `/usr/include/`, since the `-I` flag is being used anyways to add the include path, the include path that gets used can just be the `include` directory in the submodule.

Reply:
Got it, thanks for your good suggestions. I will apply them in next diff.

Reply:
Fixed, would you please have a check again.